packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/calibrate-operation.ts

```diff
@@ -26,6 +26,9 @@ export interface CalibrationOperationStateCiemss extends BaseState {
 	endTime: number;
 	stepSize: number;
 	learningRate: number;
+
+	// This is a state variable to indicate which model variables are selected for calibration, this is filled in from the workflow template generator
+	templateSelectedModelVariables: string[];
```

Comment:
Not a fan of this because it couples two sort of orthogonal ideas together (operators and scenarios), would prefer if we can build scenarios on top of operators rather than beside (or within) operators.

Maybe we can have a chat and brainstorm about this

Reply:
Ya that sounds good

Reply:
We checked in with gauldie - it is probably better to split this into two steps than try to around both returning results and exception handling.
- cleaned equation
- the skema stuff

Reply:
I think this should all be addressed now after our discussion, I am now setting the chartSettings state and removed the `templateSelectedModelVariables` entirely